### DISCUSSION TITLE
SB will not start when I add 'xvfb=True'

### DISCUSSION THREAD

**Discussion by ScottCov** [seleniumbase/SeleniumBase#3123] [General]
If I don't omit  xvfb=True, it just hangs the browser will not launch. When I do omit it, it won't click the google re-captcha at the end:

```
with SB(uc=True, headed=True, xvfb=True) as driver:

    url = "https://cases.ra.kroll.com/RMIT/Home-Index"
    driver.uc_open_with_reconnect(url, 4)
    driver.uc_click('#CASEINFO > a',by="css selector",timeout=30)
    driver.uc_click('#DOCKET > a',by="css selector",timeout=30)
    driver.uc_click('#btnAdvanced',by="css selector",timeout=30)
    sttr='190,195,196,197,236,251,252,261,307,308,309,310,311,312,313,314,315,316,326,335,352,455,457,522,523,524,531,532,592,601,602,612,617,618,619,627,628,663,674,675,676,677,699,700,730,734,736,739,745,746,747,748,749,754,766,767,776,780,791,793,795,801,812,830,831,832,843,847,848,853,863,864,866,868,869,877,881,897,898,901,906,913,919,921,922,924,925,938,944,946,948,959,963'
    item = driver.find_element('[id="txtDocketNumber"]')
    item.send_keys(sttr)
    time.sleep(10)
    driver.uc_click('#search-button',by="css selector",timeout=30)
    time.sleep(10)
    driver.uc_click('[id="1"] span [class="link"]',by="css selector",timeout=30)
    time.sleep(10)
    driver.uc_gui_click_captcha()
    time.sleep(30)
```

Am I missing a step here?

Debian 12 
Python 3.10.8
seleniumbase 4.30.3

**Comment by mdmintz** [comment]
`xvfb=True` is only the default option on Linux if you're not changing `headed`, `headless`, or `headless2` args. If you set any of those three, then you need to directly specify `xvfb=True`. This gives you some flexibility when running on Linux. For instance, some people have Linux computers with a GUI, and so we want ways of running scripts on those machines without the browser being hidden away in a virtual display.

UC Mode is not designed for reCAPTCHA. reCAPTCHA is different from other anti-bot services because it throws CAPTCHAs even if not detecting Selenium. Instead of bypassing reCAPTCHA, solve the audio challenge with an external repo:
https://github.com/search?q=pydub.AudioSegment.from_mp3+recaptcha+solver+language%3APython&type=code

**Comment by ScottCov** [comment]
And for the avoidance of doubt, I am getting this one:

![image](https://github.com/user-attachments/assets/de34380a-0daa-4739-9ada-099a7d7f13ec)

This is a reCAPTCHA, correct?

**Reply by mdmintz** [reply]
That's a reCAPTCHA. UC Mode does **not** work on it.

**Reply by ScottCov** [reply]
OK but after I click "I am not a robot" it seldomly asked me to solve a puzzle...can I automate just the clicking process without detection? And by utilizing UC mode (which I am) is it less likely to ask me to solve a puzzle?
